homework/81621/syntax_highlighter/src/lexer/tokenizer.rs

```diff
@@ -0,0 +1,596 @@
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == ';' {
+                self.tokens.push(Token {
+                    r#type: TokenType::Semicolon,
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == ',' {
+                self.tokens.push(Token {
+                    r#type: TokenType::Comma,
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == '_' {
+                self.tokens.push(Token {
+                    r#type: TokenType::Underscore,
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == ':' {
+                self.tokens.push(Token {
+                    r#type: TokenType::Colon,
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == '-' {
+                let char_string = character.to_string();
+                chars.next();
+
+                if let Some(next_char) = chars.peek() {
+                    if *next_char == '>' {
+                        self.tokens.push(Token {
+                            r#type: TokenType::SmallArrow,
+                            string: Some("->".to_string()),
+                            number: None,
+                        });
+
+                        chars.next();
+                        continue;
+                    }
+                }
+
+                self.tokens.push(Token {
+                    r#type: TokenType::Minus,
+                    string: Some(char_string),
+                    number: None,
+                });
+            } else if *character == '\'' {
+                let char_string = Tokenizer::read_char_string(&mut chars);
+
+                self.tokens.push(Token {
+                    r#type: TokenType::CharString,
+                    string: Some(char_string),
+                    number: None,
+                });
+            } else if *character == '+' {
+                self.tokens.push(Token {
+                    r#type: TokenType::Plus,
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == '*' {
+                self.tokens.push(Token {
+                    r#type: TokenType::Multiply,
+                    string: Some(character.to_string()),
+                    number: None,
+                });
+
+                chars.next();
+            } else if *character == '/' {
```

Comment:
It seems like half the code here is handling a single character operator or token, can you avoid duplicating all that in rust?